### DISCUSSION TITLE
pg types enforce users to define query config values as array of any

### DISCUSSION THREAD

**Discussion by kovacsandor** [DefinitelyTyped/DefinitelyTyped#68869] [Issues with a @types package]
With the current types if you want to wrap the query method of the `pg.Client` into your own method and also want to pass the `I` (query config values) generic parameter, you are forced to use `I extends any[]`. 

But this is not necessary, there is another way to enforce the array (and tuple) type. 

```
import {
  Client,
  ClientConfig,
  QueryConfig,
  QueryResult,
  QueryResultRow,
} from "pg";

export class DatabaseConnector {
  private client: Client;

  constructor(config: ClientConfig) {
    this.client = new Client(config);
  }

  public async connect(): Promise<void> {
    await this.client.connect();
  }

  public async end(): Promise<void> {
    await this.client.end();
  }

  public async query<T extends QueryResultRow, I extends any[]>(
    queryConfig: QueryConfig<I>,
  ): Promise<QueryResult<T>> {
    return this.client.query<T, I>(queryConfig);
  }
}
```

[a comment by an automated account is not shown]

**Comment by kovacsandor** [comment]
Fixed by https://github.com/DefinitelyTyped/DefinitelyTyped/pull/68871

**Reply by hakyonglee** [reply]
@kovacsandor 
Thank you for your contribution.

but i think there is an some problems in your fix.
i posted an issue. please check this issue.

#69248
